src/components/icon.js

```diff
@@ -5,20 +5,78 @@
   orientation: 'VERTICAL',
   jsx: (() => {
     const { Icons } = window.MaterialUI;
+    const { Link, Badge } = window.MaterialUI.Core;
+    const { useText } = B;
     const isDev = B.env === 'dev';
-    const { icon } = options;
+    const {
+      icon,
+      addBadge,
+      content,
+      badgeColor,
+      invisible,
+      anchorOrigin,
+      variant,
+      linkTo,
+      linkToExternal,
+      linkType,
+    } = options;
+
+    const hasLink = linkTo && linkTo.id !== '';
+    const hasExternalLink = linkToExternal && linkToExternal.id !== '';
+
+    const Content = useText(content);
+    const anchorOriginSplit = anchorOrigin.split(',');
+    const anchorOriginObj = {
+      horizontal: anchorOriginSplit[0],
+      vertical: anchorOriginSplit[1],
+    };
+
     const IconComponent = React.createElement(Icons[icon], {
       className: classes.root,
     });
 
+    const LinkComponent = (
+      <Link
+        href={
+          linkType === 'external' && hasExternalLink
+            ? linkToExternal
+            : 'https://google.com'
+        }
+        component={linkType === 'internal' && hasLink ? B.Link : undefined}
+        endpoint={linkType === 'internal' && hasLink ? linkTo : undefined}
+      >
+        {IconComponent}
+      </Link>
+    );
+
+    const Icon = hasLink ? LinkComponent : IconComponent;
+
+    const BadgeComponent = (
+      <div className={classes.badge}>
```

Comment:
Why this extra DIV? Can't you just use the Badge component itself for styling? To override styling in a Material component you can this: https://material-ui.com/api/badge/#css.
Then for example when you want to override the root class, you can do something like this:
```
<Badge
  classes={{ root: classes.badge }}
  badgeContent={Content}
  color={badgeColor}
  invisible={invisible}
  anchorOrigin={anchorOriginObj}
  variant={variant}
>
```

Reply:
Fixed